python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -1244,3 +1356,113 @@ def disjoint(self, other, align=True):
             align=align
         )
         return predicate(self, other)
+
+    def touches(self, other, align=True):
+        """Returns True for all aligned geometries that touch other, else
+        False.
+
+        Geometries touch if they have at least one point in common, but their
+        interiors do not intersect.
+
+        Parameters
+        ----------
+        other
+            a cuspatial.GeoSeries
+        align=True
+            align the GeoSeries indexes before calling the binpred
+
+        Returns
+        -------
+        result : cudf.Series
+            A Series of boolean values indicating whether each geometry
+            touches the corresponding geometry in the input."""
+        predicate = TOUCHES_DISPATCH[(self.column_type, other.column_type)](
+            align=align
+        )
+        return predicate(self, other)
+
+    def _basic_equals(self, other):
+        """Utility method that returns True if any point in the lhs geometry
+        is equal to a point in the rhs geometry."""
+        lhs = _multipoints_from_geometry(self)
+        rhs = _multipoints_from_geometry(other)
+        result = pairwise_multipoint_equals_count(lhs, rhs)
+        return result > 0
+
+    def _basic_equals_all(self, other):
+        """Utility method that returns True if all points in the lhs geometry
+        are equal to points in the rhs geometry."""
+        lhs = _multipoints_from_geometry(self)
+        rhs = _multipoints_from_geometry(other)
+        result = pairwise_multipoint_equals_count(lhs, rhs)
+        sizes = (
+            lhs.multipoints.geometry_offset[1:]
+            - lhs.multipoints.geometry_offset[:-1]
+        )
+        return result == sizes
+
+    def _basic_equals_count(self, other):
+        """Utility method that returns the number of points in the lhs geometry
+        that are equal to a point in the rhs geometry."""
+        lhs = _multipoints_from_geometry(self)
+        rhs = _multipoints_from_geometry(other)
+        result = pairwise_multipoint_equals_count(lhs, rhs)
+        return result
+
+    def _basic_intersects_pli(self, other):
+        """Utility method that returns the original results of
+        `pairwise_linestring_intersection`."""
+        from cuspatial.core.binops.intersection import (
+            pairwise_linestring_intersection,
+        )
+
+        lhs = _linestrings_from_geometry(self)
+        rhs = _linestrings_from_geometry(other)
+        return pairwise_linestring_intersection(lhs, rhs)
+
+    def _basic_intersects_count(self, other):
```

Comment:
Do all these internal `_basic...` methods need to be in GeoSeries? Might be a bit cluttering when trying to read the GeoSeries implementation.   They mostly don't access member data directly. Could they be in binpred_utils like `_multipoints_from_geometry`?

Reply:
Yeah I've struggled with this location and you and Michael also both thought that it wasn't clean enough. I've moved them all to `binpreds/basic_predicates.py`.